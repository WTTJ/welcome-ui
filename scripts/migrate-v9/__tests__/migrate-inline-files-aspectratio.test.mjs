import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('Inline Migration - AspectRatio Components', () => {
  const tempDir = setupMigrationTest('inline-aspectratio-test', '__fixtures__')
  it('migrates AspectRatio components with styling props to AspectRatio with classNames', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'AspectRatioComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('aspectratio-component.tsx')
  })
})
