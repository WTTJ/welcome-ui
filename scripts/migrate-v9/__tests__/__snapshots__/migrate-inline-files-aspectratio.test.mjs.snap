// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - AspectRatio Components > migrates AspectRatio components with styling props to AspectRatio with classNames > aspectratio-component.tsx 1`] = `
"import { AspectRatio } from 'welcome-ui/AspectRatio'
import { Box } from 'welcome-ui/Box'

export const AspectRatioComponent = () => {
  return (
    <div>
      <AspectRatio className="flex-1 max-h-[10rem] md:max-h-[5rem] max-w-full md:max-w-[8rem]">
        <img
          alt="video_thumbnail"
          className="rounded-md"
          loading="lazy"
          src="https://via.placeholder.com/160"
        />
      </AspectRatio>
    </div>
  )
}
"
`;
